CMakeLists.txt

```diff
@@ -684,6 +680,19 @@ if(NOT IOS AND NOT XROS)
   include(cmake/OpenCVDetectPython.cmake)
 endif()
 
+# Use same flags for native AArch64 and RISC-V compilation as for cross-compile (Linux)
+if((CV_GCC OR CV_CLANG) AND UNIX AND NOT APPLE AND NOT CMAKE_CROSSCOMPILING AND NOT CMAKE_TOOLCHAIN_FILE)
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64") # Maybe use AARCH64 variable?
+    include(${CMAKE_CURRENT_LIST_DIR}/platforms/linux/flags-aarch64.cmake)
```

Comment:
> platforms/linux

OSX/iOS?
android?
other OS?

We should not depend on "linux" here.

Reply:
This block's purpose is to use the same flags for native Linux compilation as for cross-compilation with **our** toolchain files.
* It is not applicable to Android, because it uses own toolchain files from NDK and doesn't have native compilation (or does it?).
* Also not applicable to OSX/iOS because we rely on CMake/XCode to set platform flags for specific target. 
* Not so sure about Windows, but it was not handled specifically by OpenCVCompilerOptimizations.cmake (we have only GCC/Clang flags there for AArch64). 
* Other platforms are not officially supported by us and I think we should recommend users to use there own cmake-toolchain files  for these cases.